odfuzz/generators.py

```diff
@@ -3,120 +3,223 @@
-    def generate():
-        return 'true' if random.random() < 0.5 else 'false'
 
+    @staticmethod
+    def generate(format='uri'):
+        if format == 'body':
+            return True if random.random() < 0.5 else False
+        elif format == 'uri':
+            return 'true' if random.random() < 0.5 else 'false'
+        elif format == 'key':
+            if random.random() < 0.5:
+                return "true", True
+            else:
+                return "false", False
+        else:
+            raise ValueError
 
 class EdmByte:
     @staticmethod
-    def generate():
-        return str(round(random.randint(0, 255)))
+    def generate(format='uri'):
+        value = str(round(random.randint(0, 255)))
+        if format == 'uri' or format == 'body':
+            return value
+        elif format == 'key':
+            return value, value
+        else:
+            raise ValueError
 
 
 class EdmDateTime:
+
     @staticmethod
-    def generate():
+    def generate(format='uri'):
         """
         The format of Edm.DateTime is defined as datetime'yyyy-mm-ddThh:mm[:ss[.fffffff]]'. The attribute Precision,
         which is used for declaring a microsecond as a decimal number, is ignored.
         """
-        random_date = START_DATE + datetime.timedelta(seconds=random.randint(0, DATE_INTERVAL))
-        return 'datetime\'{0}\''.format(datetime.datetime.strftime(random_date, '%Y-%m-%dT%I:%M:%S'))
+        body_value = random.randint(0, DATE_INTERVAL)
+        random_date = START_DATE + datetime.timedelta(seconds=body_value)
+        uri_value = 'datetime\'{0}\''.format(datetime.datetime.strftime(random_date, '%Y-%m-%dT%I:%M:%S'))
+        if format == 'uri':
+            return uri_value
+        elif format == 'body':
+            return body_value
+        elif format == 'key':
+            return uri_value, body_value
+
+
 
 
 class EdmDecimal:
     @staticmethod
-    def generate(self):
+    def generate(self,format='uri'):
         divider = random.randint(1, 10 ** self.scale)
         scale_range = random.randint(0, self.scale)
         rand_int = random.randint(1, (10 ** (self.precision - scale_range)) - 1)
-        return '{0:.{1}f}'.format(rand_int / divider, scale_range) + 'm'
-
+        value = '{0:.{1}f}'.format(rand_int / divider, scale_range) + 'm'
+        if format == 'uri' or format == 'body':
+            return value
+        elif format == 'key':
+            return value , value
+        else:
+            raise ValueError
 
 class EdmDouble(EncoderMixin):
     @staticmethod
-    def generate():
+    def generate(format='uri'):
         random_double = '{}d'.format(round(random.uniform(2.23e-40, 1.19e+40), 15))
-        return EdmDouble._encode_string(random_double)
+        value = EdmDouble._encode_string(random_double)
+        if format == 'uri' or format == 'body':
+            return value
+        elif format == 'key':
+            return value , value
+        else:
+            raise ValueError
```

Comment:
In the future perhaps. At the moment, we want to keep simple to read, even with duplications.  I would expect this to be covered in some overall cleanup of entire codebase. generators.py is not the most problematic module anyway and it will be easy to do :)